commit 65de9d2c218bf4356a9399b0a695322e6bca76a7 (refs/changes/05/2505/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-09-06T14:48:01-07:00 (2 years, 1 month ago)
    
    CNAPI-721 CNAPI passing to node-moray `limit` and `offset` as strings

diff --git a/lib/endpoints/servers.js b/lib/endpoints/servers.js
index b0e58c9..42a30f1 100644
--- a/lib/endpoints/servers.js
+++ b/lib/endpoints/servers.js
@@ -112,8 +112,8 @@ Server.list = function handlerSeverList(req, res, next) {
         return;
     }
 
-    if (req.params.hasOwnProperty('limit')) {
-        var limit = Number(req.params.limit); // TODO: int guard, do it above?
+    if (req.params.limit !== undefined) {
+        var limit = Number(req.params.limit);
         if (limit < SERVER_LIST_MIN_LIMIT || SERVER_LIST_MAX_LIMIT < limit) {
             res.send(400, validation.formatValidationErrors([ {
                 param: 'limit',
@@ -140,11 +140,12 @@ Server.list = function handlerSeverList(req, res, next) {
             options.default = false;
 
             // Set up paging
-            ['limit', 'offset'].forEach(function (f) {
-                if (req.params.hasOwnProperty(f)) {
-                    options[f] = req.params[f];
-                }
-            });
+            if (req.params.limit !== undefined) {
+                options.limit = Number(req.params.limit);
+            }
+            if (req.params.offset !== undefined) {
+                options.offset = Number(req.params.offset);
+            }
 
             // Set up extras
             if (req.params.extras) {
diff --git a/lib/models/server.js b/lib/models/server.js
index 12716bb..be310fb 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -133,6 +133,9 @@ ModelServer.createComputeNodeAgentHandler = function (app, jobuuid) {
  */
 
 ModelServer.list = function (params, callback) {
+    assert.optionalNumber(params.limit, 'params.limit');
+    assert.optionalNumber(params.offset, 'params.offset');
+
     var self = this;
 
     var uuid = params.uuid;
@@ -204,11 +207,12 @@ ModelServer.list = function (params, callback) {
             }
         };
 
-    ['limit', 'offset'].forEach(function (f) {
-        if (params.hasOwnProperty(f)) {
-            findOpts[f] = params[f];
-        }
-    });
+    if (params.limit !== undefined) {
+        findOpts.limit = params.limit;
+    }
+    if (params.offset !== undefined) {
+        findOpts.offset = params.offset;
+    }
 
     var servers = [];
     var serverStatus = {};
